README.md

```diff
@@ -17,60 +17,84 @@ CodeQL populates the `suppression` property in its SARIF output based on the res
 ### Example - CodeQL 
 
 ```yaml
-name: "CodeQL"
+name: "CodeQL Advanced"
 
 on:
   push:
-    branches: [ main ]
+    branches: [main]
   pull_request:
-    branches: [ main ]
-
+    branches: [main]
+  schedule:
+    - cron: "31 7 * * 3"
 jobs:
   analyze:
-    name: Analyze
-    runs-on: ubuntu-latest
+    name: Analyze (${{ matrix.language }})
+    runs-on: ${{ (matrix.language == 'swift' && 'macos-latest') || 'ubuntu-latest' }}
     permissions:
+      security-events: write
+      packages: read
       actions: read
       contents: read
-      security-events: write
 
     strategy:
       fail-fast: false
       matrix:
-        language: [ "java" ]
+        include:
+        - language: go
+          build-mode: autobuild
+        - language: java-kotlin
+          build-mode: none 
+        - language: javascript-typescript
+          build-mode: none
+        - language: python
+          build-mode: none
 
     steps:
     - name: Checkout repository
-      uses: actions/checkout@v3
+      uses: actions/checkout@v4
+
+    - name: Map Languages
```

Comment:
It's a bit unfortunate that this is necessary. The following
```
packs: "codeql/${{ env.language }}-queries:AlertSuppression.ql"
```
to run the `AlertSuppression.ql` query of the right language pack was quite a handy one-liner. However, it's not much of a one-liner anymore now it requires an additional 8 line step :-( 

Perhaps it would be clearer to add a `query` property to the matrix and write the example as follows. It's a little repetitive, but I think it is easier to understand.

```yaml
      matrix:
        include:
        - language: go
          build-mode: autobuild
          query: codeql/go-queries:AlertSuppression.ql
        - language: java-kotlin
          build-mode: none 
          query: codeql/java-queries:AlertSuppression.ql
        - language: javascript-typescript
          build-mode: none
          query: codeql/javascript-queries:AlertSuppression.ql
        - language: python
          build-mode: none
          query: codeql/python-queries:AlertSuppression.ql

      uses: github/codeql-action/init@v3
      with:
        languages: ${{ matrix.language }}
        build-mode: ${{ matrix.build-mode }}
        # run an 'alert-suppression' query
        packs: ${{ matrix.query }}
```